### PR TITLE
Mobx package upgrade

### PR DESCRIPTION
**Pull request by ixrock**
- "mobx": "_^6.3.7_",
- "mobx-observable-history": "^_2.0.3_", (includes @Nokel81 's fix https://github.com/ixrock/mobx-observable-history/pull/8)

closes lensapp/lens/pull/4135

### CODE REVIEW

File: src/renderer/components/delete-cluster-dialog/delete-cluster-dialog.tsx [hunk 1]
```diff
@@ -52,15 +52,12 @@ type Props = {};
 
 @observer
 export class DeleteClusterDialog extends React.Component {
-  showContextSwitch = false;
-  newCurrentContext = "";
+  @observable showContextSwitch = false;
```
Comment (ixrock) [inline src/renderer/components/delete-cluster-dialog/delete-cluster-dialog.tsx:55]: This file changes fixes a new error from `mobx`:

```
fix: mobx.esm.js:87 Uncaught Error: [MobX] makeObservable second arg … …
e69655b
…must be nullish when using decorators. Mixing @decorator syntax with annotations is not supported.
```

So, it seems like from this PR we have to stick to single style of annotations within a class? 
Use decorators or not (_yes_).

Comment (Nokel81) [reply src/renderer/components/delete-cluster-dialog/delete-cluster-dialog.tsx:55]: I think we use decorators more often so at least for the time being we should stick with that majority.

PR looks good, just going to test it and then approve.

**Review by Nokel81 — APPROVED** [review]

**Comment by Nokel81** [comment]
Actually please don't merge yet, I cannot seem to open drawers

**Review by Nokel81 — CHANGES_REQUESTED** [review]
Cannot open drawers with this change

**Comment by jim-docker** [comment]
I'm wondering if we should do this in 5.3.1 (i.e. don't commit to master yet). Something like this might need some time to soak/test and we may not have that time before 5.3.0 is released?

**Comment by Nokel81** [comment]
@ixrock would making `mobx` a "peerDependency" of your package be a good idea, maybe this is caused by a version mismatch.

File: src/common/configure-packages.ts [hunk 2]
```diff
@@ -29,7 +29,6 @@ export default function configurePackages() {
   // Docs: https://mobx.js.org/configuration.html
   Mobx.configure({
     enforceActions: "never",
-    isolateGlobalState: true,
```
Comment (ixrock) [inline src/common/configure-packages.ts:32]: Seems like we don't need this: https://mobx.js.org/configuration.html#isolateglobalstate-boolean

**Comment by ixrock** [comment]
@Nokel81 PTALA

**Review by Nokel81 — APPROVED** [review]

**Comment by Nokel81** [comment]
> I'm wondering if we should do this in 5.3.1 (i.e. don't commit to master yet). Something like this might need some time to soak/test and we may not have that time before 5.3.0 is released?

I think I agree that we should hold off until after the release of 5.3.0

**Pull request merged**